x-pack/plugins/security_solution/common/detection_engine/get_query_filter.ts

```diff
@@ -31,24 +31,44 @@ export const getQueryFilter = (
     title: index.join(),
   };
 
-  const initialQuery = [{ query, language }];
+  const config: EsQueryConfig = {
+    allowLeadingWildcards: true,
+    queryStringOptions: { analyze_wildcard: true },
+    ignoreFilterIfFieldNotInIndex: false,
+    dateFormatTZ: 'Zulu',
+  };
+
+  const enabledFilters = ((filters as unknown) as Filter[]).filter((f) => !isFilterDisabled(f));
```

Comment:
It might be good to have some extra validation here. We could be explicit about which fields we need `Filter` to contain, since it's so loosely defined atm. At this point, `enabledFilters` could really be anything and we won't have a good idea of why things break, if they do. I know this PR is just modifying existing code, but maybe could look at this as a follow-up.

Reply:
Yeah that would be better. I know getQueryFilter is used in a few different places, I haven't looked into why it needs to pass in `Partial<Filter> instead of a complete `Filter`, but it seems like something to fix.